worker/upload_test.go

```diff
@@ -125,6 +125,14 @@ func TestUploadDownload(t *testing.T) {
 	if !errors.Is(err, errBucketNotFound) {
 		t.Fatal("expected bucket not found error", err)
 	}
+
+	// upload data using a cancelled context - assert we don't hang
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	_, _, err = ul.Upload(ctx, bytes.NewReader(data), metadatas, params, lockingPriorityUpload)
```

Comment:
This is a good case to have overall but it's not going to catch the bug that this PR fixes.
If you want you can f/u with a regression test so that we can test the fix in the meantime on a node though.

A possible regression test could start an upload with no memory left so that it gets blocked in the right spot, then wait a second or so for it to reach that spot and finally close the context.

Reply:
I'll keep the case because this already hangs the upload. I'll add a regression test in a F/U.